src/GitVersionCore.Tests/ExecuteCoreTests.cs

```diff
@@ -40,6 +42,35 @@ public void CacheKeySameAfterReNormalizing()
         });
     }
 
+    [Test]
+    public void CacheKeyForWorktree() {
+        var versionAndBranchFinder = new ExecuteCore(fileSystem);
+
+        if (Type.GetType("Mono.Runtime") != null)
+            Assert.Ignore("LibGit2Sharp fails here when running under Mono");
```

Comment:
I think there might be tests that already ignore Mono through the use of `[Category]` or something similar. Can you please investigate that option?

Reply:
I did see the `NoMono` category, but it looks like the logic is to [skip the whole test when running on Unix](https://github.com/GitTools/GitVersion/blob/f9532574212bc48db5c8d099ceec05ec13bb55da/build/pack.cake#L66-L69). This test can run with .NET Core, so I didn't want to skip it altogether.

Reply:
Aha, that logic is wrong, then. There are surely a lot of tests decorated with `NoMono` that break on Unix, but then they should be marked with a `NoUnix` category. To not continue digging in this erroneous hole and make it any bigger, let's fix the `NoMono` vs. `NoUnix` issue first and then decorate this test with `NoUnix`. Ok? If you want to tackle it in this PR that would be very much appreciated, otherwise I can take a look in another PR if I can find a few minutes of spare time.

Reply:
I started adding the changes for a NoUnix category, but there were some required category changes so I think its better in a separate PR. I'll open another one once I work through the failing tests.

For now, I've just switched my new test over to NoMono so it should pass.